packages/gatsby/src/query/index.js

```diff
@@ -72,11 +72,7 @@ const popNodeQueries = state => {
 
     // Find components that depend on this node so are now dirty.
     if (state.componentDataDependencies.nodes.has(node.id)) {
-      state.componentDataDependencies.nodes.get(node.id).forEach(n => {
-        if (n) {
-          dirtyIds.add(n)
-        }
-      })
+      state.componentDataDependencies.nodes.get(node.id).forEach(dirtyIds)
```

Comment:
I think this would need to `.forEach(dirtyIds.add)` (there was missing actual `.add` method (binding might also be necessary)

Reply:
Ah, missed that yeah... I've just pushed it with .add and explicitly binding the n

Reply:
Yeah it works now 🙃